pkg/controller/grouping/group_entity_index.go

```diff
@@ -328,6 +348,39 @@ func (i *GroupEntityIndex) AddNamespace(namespace *v1.Namespace) {
 	}
 }
 
+func (i *GroupEntityIndex) AddNode(node *v1.Node) {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	nodeLabels, exists := i.nodeLabels[node.Name]
+	// Do nothing if labels are not updated.
+	if exists && labels.Equals(nodeLabels, node.Labels) {
+		return
+	}
+
+	i.nodeLabels[node.Name] = node.Labels
+}
+
+func (i *GroupEntityIndex) UpdateNode(node *v1.Node) {
+	i.lock.Lock()
+	defer i.lock.Unlock()
+
+	nodeLabels, exists := i.nodeLabels[node.Name]
+	// Do nothing if labels are not updated.
+	if exists && labels.Equals(nodeLabels, node.Labels) {
+		return
+	}
+
+	i.nodeLabels[node.Name] = node.Labels
+}
```

Comment:
Don't we have to scan the `nodeLabelItems`, notify the selectorItems and trigger the update chain? Refer to AddPod or AddNamespace.